apps/designer/app/designer/features/sidebar-left/panels/pages/pages.tsx

```diff
@@ -60,36 +67,63 @@ const staticTreeProps = {
   getIsExpanded(_node: PagesTreeNode) {
     return true;
   },
-  renderItem(props: { data: PagesTreeNode; isSelected: boolean }) {
-    if (props.data.type === "folder") {
-      return null;
-    }
-
-    return (
-      <>
-        <TreeNodeLabel
-          isSelected={props.isSelected}
-          text={props.data.data.name}
-        />
-      </>
-    );
-  },
 };
 
 const PagesPanel = ({
   onClose,
   onNewPage,
   onSelect,
   selectedPageId,
+  onEdit,
+  editingPageId,
 }: {
   onClose?: () => void;
   onNewPage?: () => void;
   onSelect: (pageId: string) => void;
   selectedPageId: string;
+  onEdit?: (pageId: string) => void;
+  editingPageId?: string;
 }) => {
   const [pages] = usePages();
   const pagesTree = useMemo(() => pages && toTreeData(pages), [pages]);
 
+  const renderItem = useCallback(
```

Comment:
maybe this should be a component Item?

Reply:
Sorry, don't understand what you mean :)

Reply:
Ah, you mean instead of passing `renderItem` function to the tree, pass a component?

I think that we'll be slightly worse as we would have to create components dynamically during render, etc. But overall I don't have a strong opinion.

Reply:
Keeping as is unless there're reasons to go with component